docs/learn/migrate/solana.mdx

```diff
@@ -1,8 +1,689 @@
+    Ok(())
+}
+```
+
+**After (Stellar):**
+
+```rust
+use soroban_sdk::{
+    contract, contractimpl, symbol_short, vec, Address, Env, Symbol, Vec,
+};
+
+#[derive(Clone, Debug, Eq, PartialEq, soroban_sdk::serde::Serialize, soroban_sdk::serde::Deserialize)]
+pub struct UserData {
+    pub name: Symbol,
+    pub balance: u64,
+    pub is_active: bool,
+}
+
+#[contract]
+pub struct UserDataContract;
+
+#[contractimpl]
+impl UserDataContract {
+    pub fn create_user(env: Env, user: Address, name: Symbol) {
+        let user_data = UserData {
+            name,
+            balance: 0,
+            is_active: true,
+        };
+        env.storage().instance().set(&user, &user_data);
+    }
+
+    pub fn update_balance(env: Env, user: Address, amount: u64) {
+        if let Some(mut user_data) = env.storage().instance().get(&user) {
+            user_data.balance += amount;
+            env.storage().instance().set(&user, &user_data);
+        }
+    }
+
+    pub fn get_user_data(env: Env, user: Address) -> Option<UserData> {
+        env.storage().instance().get(&user)
+    }
+}
+```
+
+The Stellar version is much cleaner - no manual serialization/deserialization, and the storage system automatically handles the data lifecycle.
+
+## Advanced Migration Patterns
+
+Now let's look at some more complex patterns you might encounter when migrating larger contracts.
+
+### Cross-Contract Invocation
+
+Cross-contract calls work differently between the two platforms.
+
+**Before (Solana):**
+
+```rust
+use solana_program::{
+    program::invoke,
+    system_instruction,
+};
+
+pub fn process(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    instruction_data: &[u8],
+) -> ProgramResult {
+    let accounts_iter = &mut accounts.iter();
+    let payer = next_account_info(accounts_iter)?;
+    let recipient = next_account_info(accounts_iter)?;
+    let system_program = next_account_info(accounts_iter)?;
+
+    // Transfer SOL
+    let transfer_instruction = system_instruction::transfer(
+        &payer.key,
+        &recipient.key,
+        1000000, // 0.001 SOL
+    );
+
+    invoke(
+        &transfer_instruction,
+        &[payer.clone(), recipient.clone(), system_program.clone()],
+    )?;
+
+    Ok(())
+}
+```
+
+**After (Stellar):**
+
+```rust
+use soroban_sdk::{
+    contract, contractimpl, symbol_short, vec, Address, Env, Symbol, Vec,
+};
+
+#[contract]
+pub struct TransferContract;
+
+#[contractimpl]
+impl TransferContract {
+    pub fn invoke_other_contract(env: Env, contract_id: Address, function: Symbol, args: Vec<Symbol>) {
+        // Invoke another contract
+        let _: Vec<Symbol> = env.invoke_contract(&contract_id, &function, &args);
+    }
+}
+```
+
+:::note **Important**:
+
+Stellar's `env.invoke_contract()` is the low-level way to call other contracts. For better type safety and developer experience, you can also use the `contractimport!` macro to generate typed clients for contract interactions.
```

Comment:
Instead of the note, couldn't we demonstrate the usual, high-level way of invoking the contracts via `contractimport!` (like we do e.g. in https://github.com/stellar/soroban-examples/blob/main/cross_contract/contract_b/src/lib.rs), and add a note about the existence of a lower level function? 
If I understand the example above correctly, you want to perform a token transfer. That's easy to do in Stellar as well, and you don't even need to import a separate contract, you can just use `soroban_sdk::token::Client` instead to make a `transfer` call.